src/NLog.Targets.Redis/NLog.Targets.Redis.csproj

```diff
@@ -30,12 +30,16 @@
 
   <ItemGroup Condition=" '$(TargetFramework)' == 'net45' ">
     <PackageReference Include="NLog" Version="4.0.0" />
-    <PackageReference Include="StackExchange.Redis.StrongName" Version="1.0.488" />
   </ItemGroup>
 
   <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.5' ">
     <PackageReference Include="NLog" Version="4.5.0" />
     <PackageReference Include="StackExchange.Redis.StrongName" Version="1.1.605" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Include="NLog" Version="4.7.4" />
+    <PackageReference Include="StackExchange.Redis" Version="2.1.58" />
```

Comment:
Was the reason for the need for this upgrade being the usage of the strong name version of StackExchange.Redis? If there are no breaking changes, then we should use the lowest version possible.

Reply:
Also, we should use the lowest version of NLog possible. We don't want to force the upgrade of NLog.

Reply:
FYI, nowadays at NLog we also force upgrades. (See for example NLog.AspNetCore), I think there are 2 types of users, some who are doing updates and others who don't care.

For the best performance, security and stability, limiting the combinations (e.g. old with new), is the best IMO

Reply:
I can see that being a good practice with NLog, but I worry about packages shared with the application - e.g. StackExchange.Redis. Since commonly StackExchange.Redis is used by the application proper.

StackExchange.Redis pulls in a lot of dependencies, so by referencing StackExchange.Redis we implicitly reference things like `Microsoft.Win32.Registry 4.7.0` or `System.Runtime.CompilerServices.Unsafe 4.5.3`. Each could have a breaking change we don't know about. The developer may not be ready or able to update `System.Runtime.CompilerServices.Unsafe`, for example. If decoupling NLog.Targets.Redis from a version of StackExchange.Redis doesn't increase the complexity of the project, I feel we should. We should target `StackExchange.Redis >= 2.1` or `StackExchange.Redis >= 2.0` just to prevent more DLL hell, give the most flexibility with the developer.

This pr was after all to address where NLog restricted what package should be installed.

Just a real world example, we've seen Microsoft break compatibility in the past e.g. one of the Owin security packages broke serialization compatibility during a minor version change - it effectively required that every application that handled token verification be locked at a minor version until every application upgraded. I think @joshua5822 can remember that.

Reply:
Ug I remember that package I think it was something like version 3.0 being incompatible with 3.0.1+ because they forgot to increment the version number of serialized data to 3 in their 3.0 package.

In 2.0 of StackExchange.Redis they deprecated the strong name package and the main package became strong named. I can change it to target StackExchange.Redis >= 2.1 to allow more flexibility. 2.0 had a pretty rocky start due to the large amount of changes made.

Reply:
Lowest stable 2.1 package is 2.1.28, so I've down graded is to that version.

Reply:
Looks good, thanks.